zebra-state/src/lib.rs

```diff
@@ -1,341 +1,32 @@
-        let config = Config::default();
-        // we can't do many useful tests on this value, because it depends on the
-        // local environment and OS.
-        let sled_config = config.sled_config(network);
-        let mut path = sled_config.get_path();
-        assert_eq!(path.file_name(), Some(OsStr::new("state")));
-        assert!(path.pop());
-        match network {
-            Network::Mainnet => assert_eq!(path.file_name(), Some(OsStr::new("mainnet"))),
-            Network::Testnet => assert_eq!(path.file_name(), Some(OsStr::new("testnet"))),
-        }
-    }
-
-    /// Block heights, and the expected minimum block locator height
-    static BLOCK_LOCATOR_CASES: &[(u32, u32)] = &[
-        (0, 0),
-        (1, 0),
-        (10, 0),
-        (98, 0),
-        (99, 0),
-        (100, 1),
-        (101, 2),
-        (1000, 901),
-        (10000, 9901),
-    ];
-
-    /// Check that the block locator heights are sensible.
-    #[test]
-    fn test_block_locator_heights() {
-        for (height, min_height) in BLOCK_LOCATOR_CASES.iter().cloned() {
-            let locator = block_locator_heights(block::Height(height)).collect::<Vec<_>>();
-
-            assert!(!locator.is_empty(), "locators must not be empty");
-            if (height - min_height) > 1 {
-                assert!(
-                    locator.len() > 2,
-                    "non-trivial locators must have some intermediate heights"
-                );
-            }
+mod tests;
 
-            assert_eq!(
-                locator[0],
-                block::Height(height),
-                "locators must start with the tip height"
-            );
+use memory_state::MemoryState;
+use service::QueuedBlock;
+use sled_state::SledState;
 
-            // Check that the locator is sorted, and that it has no duplicates
-            // TODO: replace with dedup() and is_sorted_by() when sorting stabilises.
-            assert!(locator.windows(2).all(|v| match v {
-                [a, b] => a.0 > b.0,
-                _ => unreachable!("windows returns exact sized slices"),
-            }));
+pub use config::Config;
+pub use request::{HashOrHeight, Request};
+pub use response::Response;
+pub use service::init;
 
-            let final_height = locator[locator.len() - 1];
-            assert_eq!(
-                final_height,
-                block::Height(min_height),
-                "locators must end with the specified final height"
-            );
-            assert!(height - final_height.0 <= MAX_BLOCK_REORG_HEIGHT.0,
-                    format!("locator for {} must not be more than the maximum reorg height {} below the tip, but {} is {} blocks below the tip",
-                         height,
-                         MAX_BLOCK_REORG_HEIGHT.0,
-                         final_height.0,
-                         height - final_height.0));
-        }
-    }
-}
+/// A boxed [`std::error::Error`].
+pub type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
```

Comment:
should this be a private alias rather than part of our API?

Reply:
I made it a public alias because the type is already part of our API, so it seems convenient to expose the alias to our users as well.  This matches what we do in `zebra-network`, but we could consider removing the aliases across the workspace in a separate PR?

Reply:
Either or, I don't think it matters much what we choose so long as it wasn't done by accident.